content/6_Advanced/Lagrange.mdx

```diff
@@ -27,6 +23,144 @@ adding lambda\*smth
 	 />
 </Resources>
 
+## Lagrangian Relaxation
+
+Lagrangian Relaxation involves transforming a constraint on a variable into a cost $\lambda$ and binary searching for the optimal $\lambda$.
+
+<FocusProblem problem="sample" />
+
+The problem gives us a length $N$ ($1 \le N \le 3 \cdot 10^5$) array of integers in the range $[-10^9,10^9]$. We are given some $K$ ($1 \le K \le N$) and are asked to choose at most $K$ disjoint subarrays such that the sum of elements included in a subarray is maximized.
+
+### Intuition
+
+The main bottleneck of any dynamic programming solution to this problem is having to store the number of subarrays we have created so far.
+
+Let's try to find a way around this. Instead of storing the number of subarrays we have created so far, we assign a penalty of $\lambda$ for creating a new subarray (i.e. everytime we create a subarray we penalize our sum by $\lambda$).
```

Comment:
```suggestion
Let's try to find a way around this. Instead of storing the number of subarrays we have created so far, we assign a penalty of $\lambda$ for creating a new subarray (i.e. every time we create a subarray we penalize our sum by $\lambda$).
```